From 6479e4cd772d9bbcf7f37e6c296280ebd4b45064 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Mon, 15 Mar 2021 12:45:46 +0530
Subject: [PATCH] Changes to support external camera

- If camera device path has /, replace it with _ as / is
not allowed in file uri path.
- If no front camera found, return external camera facing
id.

Signed-off-by: shiva kumara <shiva.kumara.rudrappa@intel.com>
Signed-off-by: kbillore <kaushal.billore@intel.com>
Change-Id: I706781f8c28431357b2e0b17ca5c9c9138f19665
---
 .../processing/imagebackend/TaskCompressImageToJpeg.java  | 8 +++++++-
 src/com/android/camera/settings/SettingsManager.java      | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java b/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
index 7a2caf9fb..c12073052 100644
--- a/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
+++ b/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
@@ -420,7 +420,13 @@ public class TaskCompressImageToJpeg extends TaskJpegEncode {
      * @return Quality level to use for JPEG compression.
      */
     protected int getJpegCompressionQuality () {
-        return CameraProfile.getJpegEncodingQualityParameter(CameraProfile.QUALITY_HIGH);
+        /* Media framework doesn't support query of EXTERNAL camera quality,
+         * It will return the BACK facing camera's quality. Set quality to 90 as WA.
+         */
+        int quality = CameraProfile.getJpegEncodingQualityParameter(CameraProfile.QUALITY_HIGH);
+        if (quality == 0)
+            return 90;
+        return quality;
     }
 
     /**
diff --git a/src/com/android/camera/settings/SettingsManager.java b/src/com/android/camera/settings/SettingsManager.java
index d2794e578..80d36d93e 100644
--- a/src/com/android/camera/settings/SettingsManager.java
+++ b/src/com/android/camera/settings/SettingsManager.java
@@ -160,6 +160,7 @@ public class SettingsManager {
     }
 
     public static String getCameraSettingScope(String cameraIdValue) {
+        cameraIdValue = cameraIdValue.replaceAll("/", "_");
         return CAMERA_SCOPE_PREFIX + cameraIdValue;
     }
 
-- 
2.17.1

